backend-fsevent/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+//! Notify Backend using the Darwin FSEvents API.
+
+// For more information on the FSEvents API, the best resource is in the system
+// headers; located on your Macintosh at
+// /System/Library/Frameworks/CoreServices.framework/Frameworks/FSEvents.framework/Headers/FSEvents.h
+
+extern crate notify_backend as backend;
+extern crate libc;
+extern crate futures;
+extern crate fsevent as fsevent_rs;
+extern crate fsevent_sys;
+
+mod watcher;
+
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex, Condvar};
+use std::collections::VecDeque;
+
+use futures::{Async, Poll, Stream};
+
+use backend::prelude::*;
+use watcher::FsEventWatcher;
+
+pub type WaitQueue = Arc<(Mutex<VecDeque<Event>>, Condvar)>;
+
+pub struct Backend {
+    watcher: FsEventWatcher,
+    queue: WaitQueue,
+}
+
+impl NotifyBackend for Backend {
+    fn new(paths: Vec<PathBuf>) -> BackendResult<BoxedBackend> {
+        Ok(Box::new(Backend::new(paths)))
+    }
+
+    fn caps(&self) -> Vec<Capability> {
+        Self::capabilities()
+    }
+
+    fn capabilities() -> Vec<Capability> {
+        vec![
+            Capability::FollowSymlinks,
+            Capability::WatchEntireFilesystem,
+            Capability::WatchFiles,
```

Comment:
I've only given a cursory look to the FSEvent documentation, but it is my understanding that it cannot do that, i.e. FSEvent only watches directories (and [Apple recommends using kqueue for single files instead](https://developer.apple.com/library/content/documentation/Darwin/Conceptual/FSEvents_ProgGuide/KernelQueues/KernelQueues.html#//apple_ref/doc/uid/TP40005289-CH5-SW2)).

Reply:
With the `FileEvents` flag available from 10.7, if I create an `FSEventStream` with a single path, to a file, I will receive only events for that specific file, not events in the parent directory.

My impression is that a lot of Apple's documentation on the matter was written for the initial release, and hasn't been updated.

Reply:
Oh cool :)

Reply:
Yea, I was pleasantly surprised, because all the docs suggested this wasn't the case. :)